test/e2e/prometheus_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const monitoringNamespace string = "monitoring"
+
+// Agent metrics to validate
+var antreaAgentMetrics = []string{
+	"antrea_agent_local_pod_count",
+	"antrea_agent_ovs_flow_table",
+	"antrea_agent_runtime_info",
+}
+
+// Controller metrics to validate
+var antreaControllerMetrics = []string{
+	"antrea_controller_address_group_processed",
+	"antrea_controller_address_group_sync_duration_milliseconds",
+	"antrea_controller_address_group_sync_duration_milliseconds",
+	"antrea_controller_address_group_sync_duration_milliseconds",
+	"antrea_controller_address_group_sync_duration_milliseconds_count",
+	"antrea_controller_address_group_sync_duration_milliseconds_sum",
+	"antrea_controller_applied_to_group_processed",
+	"antrea_controller_applied_to_group_sync_duration_milliseconds",
+	"antrea_controller_applied_to_group_sync_duration_milliseconds",
+	"antrea_controller_applied_to_group_sync_duration_milliseconds",
+	"antrea_controller_applied_to_group_sync_duration_milliseconds_count",
+	"antrea_controller_applied_to_group_sync_duration_milliseconds_sum",
+	"antrea_controller_length_address_group_queue",
+	"antrea_controller_length_applied_to_group_queue",
+	"antrea_controller_length_network_policy_queue",
+	"antrea_controller_network_policy_processed",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_network_policy_sync_duration_milliseconds_count",
+	"antrea_controller_network_policy_sync_duration_milliseconds_sum",
+	"antrea_controller_runtime_info",
+}
+
+// Prometheus server JSON output
+type prometheusServerOutput struct {
+	Status string
+	Data   []map[string]string
+}
+
+// Retrieve monitoring authorization token, required for access to Antrea apiserver /metrics resource
+func getMonitoringAuthToken(t *testing.T, data *TestData) string {
+	secrets, err := data.clientset.CoreV1().Secrets(monitoringNamespace).List(metav1.ListOptions{})
```

Comment:
not directly related to this test PR, but was it good practice to use the default ServiceAccount in the monitoring namespace when deploying Prometheus, or should we have created our own ServiceAccount?

Reply:
You mean for monitoring in general?
Yeah it makes sense and worthwhile to open an issue for this.